grammars/css.cson

```diff
@@ -20,294 +20,1235 @@
+        'end': '\\)'
         'endCaptures':
-          '1':
-            'name': 'punctuation.section.function.css'
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
         'patterns': [
           {
-            'match': '[^\'") \\t]+'
-            'name': 'variable.parameter.url.css'
+            'include': '#comment-block'
           }
           {
-            'include': '#string-single'
+            'include': '#escapes'
           }
           {
-            'include': '#string-double'
+            'match': '(?i)(?<![\\w-])(ltr|rtl)(?![\\w-])'
+            'name': 'support.constant.text-direction.css'
           }
           {
-            'include': '#comment-block'
+            'include': '#property-value'
           }
         ]
       }
       {
-        'include': '#media-query-list'
-      }
-    ]
-  }
-  {
-    'begin': '((@)font-face)\\s*(?=\\{)'
-    'beginCaptures':
-      '1':
-        'name': 'keyword.control.at-rule.font-face.css'
-      '2':
-        'name': 'punctuation.definition.keyword.css'
-    'end': '(?!\\G)'
-    'name': 'meta.at-rule.font-face.css'
-    'patterns': [
-      {
-        'include': '#rule-list'
-      }
-    ]
-  }
-  {
-    'begin': '(?=@media\\b)'
-    'end': '\\s*(\\})'
-    'endCaptures':
-      '1':
-        'name': 'punctuation.section.property-list.end.css'
-    'patterns': [
-      {
-        'begin': '((@)media)\\b'
+        # Language
+        'begin': '(?i)((:)lang)(\\()'
         'beginCaptures':
           '1':
-            'name': 'keyword.control.at-rule.media.css'
+            'name': 'entity.other.attribute-name.pseudo-class.css'
           '2':
-            'name': 'punctuation.definition.keyword.css'
-        'end': '\\s*(?=\\{)'
-        'name': 'meta.at-rule.media.css'
+            'name': 'punctuation.definition.entity.css'
+          '3':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
         'patterns': [
           {
-            'include': '#media-query-list'
+            'match': '(?<=[(,\\s])[a-zA-Z]+(-[a-zA-Z0-9]*+|\\\\(?:[0-9a-fA-F]{1,6}|.))*+(?=[),\\s])'
+            'name': 'support.constant.language-range.css'
+          }
+          {
+            'begin': '"'
+            'beginCaptures':
+              '0':
+                'name': 'punctuation.definition.string.begin.css'
+            'end': '"'
+            'endCaptures':
+              '0':
+                'name': 'punctuation.definition.string.end.css'
+            'name': 'string.quoted.double.css'
+            'patterns': [
+              {
+                'include': '#escapes'
+              }
+              {
+                'match': '(?<=["\\s])[a-zA-Z*]+(-[a-zA-Z0-9*]*+)*+(?=["\\s])'
+                'name': 'support.constant.language-range.css'
+              }
+            ]
```

Comment:
I believe you were the one who added this. I left it alone because I wasn't sure what it was you were trying to do, and the possessive capturing suggested it needed to be very specific. Can you explain exactly what it was your code was trying to do?